### PR TITLE
Cxc 159 preview

### PR DESCRIPTION
**Pull request by esitsolf**
- The preview is now appearing everywhere it should (on the index when a draft is saved, on the preview popup and on the export page)
- there is no credit and border on the draft on index
- A array of promises is used to make the job
- A loader was introduce to let the user wait the preview on index
- this bug is solved: create a comic - add any assets on the comic - use the browsers back button - select another template and create it - it will still show the old comic.

### CODE REVIEW

**Comment by hannahwagner1402** [comment]
Screenshot 2024-05-21 at 09 19 09
Screenshot 2024-05-21 at 09 20 13

In our original figma design the background of the preview overlay was supposed to be transparent

File: components/PreviewCanvas.vue [hunk 1]
```diff
@@ -30,6 +30,7 @@
         // set the size of the canvas, should come from the wrapper, should be defined when choosing a template
         const stripsHeight = comicStore.comic.getPage(0).getStrip(0).height;
         const panels = comicStore.comic.getPage(0).getStrip(0).panels;
+        console.log('called', panels);
```
Comment (hannahwagner1402) [inline components/PreviewCanvas.vue]: remove console log^^

File: components/PreviewCanvas.vue [hunk 2]
```diff
@@ -107,7 +108,8 @@
         };
     }
 
-    function drawPanel(context, panel, startPoint, height) {
+    async function drawPanel(context, panel, startPoint, height) {
+        console.log('drawPanel called', panel);
```
Comment (hannahwagner1402) [inline components/PreviewCanvas.vue]: remove console log

**Comment by Annacc211018** [comment]
Text elements appear different in the editor than they do on preview and export.
image 
image

**Comment by Vanessi-K** [comment]
When the preview is active I can still scroll the page behind it.

File: pages/editor.vue [hunk 3]
```diff
@@ -313,19 +315,10 @@
...
-    .comic-preview {
-        width: 60vw;
-        height: 20vh;
-        background-color: $grey-70;
-        position: absolute;
-        top: 50%;
-        left: 50%;
-        transform: translate(-50%, -50%);
+        height: 100%;
+        background-color: $grey-60;
```
Comment (Vanessi-K) [inline pages/editor.vue:319]:
```suggestion
        background-color: rgba($grey-60, 0.4);
```

Comment (esitsolf) [reply pages/editor.vue:319]: this is not enough, but I managed to make it work :)

File: pages/editor.vue [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <script setup>
     import ComicPanels from '~/components/ComicPanels.vue';
     import iconConfig from '../config/iconsConfig';
+    import PreviewCanvas from '~/components/PreviewCanvas.vue';
```
Comment (Vanessi-K) [inline pages/editor.vue:4]: Unecessary import

**Review by Vanessi-K — CHANGES_REQUESTED** [review]
This does not look right:
image
This is in Firefox.
Looks good on my phone though.

File: pages/editor.vue (hunk 4 above)
Comment (esitsolf) [reply pages/editor.vue:4]: removed!

**Comment by Vanessi-K** [comment]
The z-index in Preview is not working correctly:
image
image

**Comment by Vanessi-K** [comment]
The preview pop-up is scrollable. Why?

**Pull request merged**